Source/Core/Core/HW/WiimoteReal/WiimoteReal.cpp

```diff
@@ -440,18 +440,25 @@ void WiimoteScanner::WantBB(bool do_want)
   m_want_bb.store(do_want);
 }
 
-void WiimoteScanner::StartScanning()
+void WiimoteScanner::StartScanning(bool scan_only_once)
 {
+  m_should_scan.store(true);
+  m_should_scan_only_once.store(scan_only_once);
```

Comment:
This should be set before triggering the WiimoteScanner thread to avoid getting wrong the value.